### DISCUSSION TITLE
Tunnel collision

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#2564] [Q&A]
Lets suppose initiator and responder receives acquire at the same time. Are both ikev2 tunnels going to established same time or is there any precaution against this collision?

**Comment by tobiasbrunner** [comment]
Depends on the timing whether the `unique` setting can prevent a duplicate. It can definitely happen that both are established, which is usually not an issue, as the uniqueness check requires an established IKE_SA with the same identities to detect a duplicate while processing IKE_AUTH.
